galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyRoute.java

```diff
@@ -0,0 +1,196 @@
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.QueryParameters;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+
+/**
+ * An abstract route used by all the Property-related routes.
+ */
+public class PropertyRoute extends CPSRoute {
+
+    static final Gson gson = GalasaGsonBuilder.build();
+
+    // Define a default filter to accept everything
+    static DirectoryStream.Filter<Path> defaultFilter = path -> { return true; };
+
+
+    public PropertyRoute(ResponseBuilder responseBuilder, IFramework framework ) {
+		/* Regex to match endpoints: 
+		*  -> /cps/
+		*/
+		super(responseBuilder, "/cps/(.*)/properties/(.*)", framework);
+	}
+
+    protected IFramework getFramework() {
+        return super.framework;
+    }
+
+    protected String getPropertyNameFromURL(String pathInfo){
+        String[] namespace = pathInfo.split("/");
+        return namespace[4];
+    }
+    
+    private Map.Entry<String, String> retrieveProperty(String namespace, String propertyName) throws  FrameworkException {
+        try{
+           Map<String, String> properties = getAllProperties(namespace);
+           for (Map.Entry<String, String> entry : properties.entrySet()) {
+               String key = entry.getKey().toString();
+               if (key.equals(propertyName)){
+                   return entry;
+               }
+           }
+        }catch (Exception e){
+            ServletError error = new ServletError(GAL5017_INVALID_NAMESPACE_ERROR,namespace);  
+            throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+        }
+        return null;
+    }
+
+    private boolean checkPropertyExists (String namespace, String propertyName) throws FrameworkException{
+        Map.Entry<String, String> entry = retrieveProperty(namespace, propertyName);
+        if( entry  == null){
+            return false;
+        }
+        return true;
+    }
+
+    private void checkRequestHasContent(HttpServletRequest request, String pathInfo) throws InternalServletException{
+        boolean valid = false;
+        try{
+            if (request.getContentLength() >0){
+                valid = true;
+            }
+        }catch (NullPointerException e ){
+            //Catch the NullPointerException (empty request body) 
+            //Exception is thrown by the if below
+        }  
+        if (!valid){
+            ServletError error = new ServletError(GAL5411_NO_REQUEST_BODY,pathInfo);  
+            throw new InternalServletException(error, HttpServletResponse.SC_LENGTH_REQUIRED);
+        }
+    }
```

Comment:
Might make more sense to return a boolean and check the request isn't null outside this method?